.github/workflows/scripts/detect_ts_ignore.py

```
@@ -0,0 +1,37 @@
+
+    Args:
+        files (list): List of file paths to check.
+
+    Returns:
+        int: 0 if no violations, 1 if violations are found.
+    """
+    ts_ignore_found = False
+
+    for file in files:
+        try:
+            with open(file, "r", encoding="utf-8") as f:
```

Ruff (0.8.2)

Unnecessary open mode parameters

Remove open mode parameters

(UP015)